docs/API-REFERENCE.md

```diff
@@ -135,29 +144,52 @@ The code should be previously registered through `jerry_parse`.
 
 ```c
 jerry_completion_code_t
-jerry_run (void);
+jerry_run (jerry_api_value_t *error_value_p);
 ```
-
-- returned value - completion code that indicates whether run performed successfully (`JERRY_COMPLETION_CODE_OK`), or an unhandled JavaScript exception occurred (`JERRY_COMPLETION_CODE_UNHANDLED_EXCEPTION`).
+- `error_value_p` - error value (output parameter)
+- returned value - completion code that indicates whether run performed successfully
+  - `JERRY_COMPLETION_CODE_OK` - successful completion
+  - `JERRY_COMPLETION_CODE_UNHANDLED_EXCEPTION` - an unhandled JavaScript exception occurred
+  - `JERRY_COMPLETION_CODE_INVALID_SNAPSHOT_VERSION` - snapshot version mismatch
+  - `JERRY_COMPLETION_CODE_INVALID_SNAPSHOT_FORMAT` - snapshot format is not valid
 
 **Example**
 
 ```c
 {
-  jerry_init (JERRY_FLAG_ENABLE_LOG);
+  const jerry_api_char_t script[] = "print ('Hello, World!');";
+  size_t script_size = strlen ((const char *) script);
 
-  char script [] = "print ('Hello, World!');";
-  jerry_parse (script, strlen (script));
+  /* Initialize engine */
+  jerry_init (JERRY_FLAG_EMPTY);
 
-  jerry_run ();
+  /* Setup Global scope code */
+  jerry_api_object_t *error_object_p = NULL;
+  if (!jerry_parse (script, script_size, &error_object_p))
+  {
+    /* Error object must be freed, if parsing failed */
+    jerry_api_release_object (error_object_p);
+  }
+  else
+  {
+    /* Execute Global scope code */
+    jerry_api_value_t error_value = jerry_api_create_void_value ();
+    jerry_completion_code_t return_code = jerry_run (&error_value);
```

Comment:
I think we should be careful when writing examples because our exaples can be misleading. So please add a comment that this initialization is not mandatory.